sympy/sets/tests/test_fancysets.py

```diff
@@ -226,37 +226,40 @@ def test_Range_set():
     assert Range(1, 10, 1)[-1] == 9
     assert Range(1, 10, 3)[-1] == 7
     raises(ValueError, lambda: Range(oo,0,-1)[1:3:0])
-    raises(ValueError, lambda: Range(oo,0,-1)[:1])
-    raises(ValueError, lambda: Range(1, oo)[-2])
-    raises(ValueError, lambda: Range(-oo, 1)[2])
+    assert Range(oo,0,-1)[:1] == empty
```

Comment:
This should raise: if you cannot iterate from the left you shouldn't be able to slice from the left.

Reply:
I made the changes according to what I interpreted from https://github.com/sympy/sympy/pull/17146#issuecomment-515649370. AFAICT, you wanted to convey that `oo` should be returned when iterating from an end having `oo`. Have I figured out something wrong. Please clarify.

Reply:
That comment was about indexing and slicing. 

Any index measured from the infinite end should either raise an error or give the infinite value so `Range(oo)[-2]` should give `oo` or else raise an error (maybe the error would be better) since one might reasonable conclude that a given slice can only correspond to 1 value so it would be weird to have `[-1]` and `[-2]` both be `oo`.

For similar reasons, *slices* measured from the infinite end should either give an unaltered result or raise an error so the behavior of `Range(oo)[:-1]` is the same as `Range(oo)[:-2]`.

It looks like instead of raising an error you were giving an empty set. Do you think that makes more sense than an error? When might this arise in practice and what should the user expect to happen?

Reply:
Ah! Okay. I got your point. I think error should be better because it will keep things unified in both slicing and indexing, i.e., in both cases we should raise error whenever we encounter an infinity while indexing/slicing.

Reply:
Updated with the commits. Please take a look.